src/vcpkg/commands.build.cpp

```diff
@@ -470,16 +470,53 @@ namespace vcpkg
         return base_env.cmd_cache.get_lazy(build_env_cmd, [&]() {
             const Path& powershell_exe_path = paths.get_tool_exe("powershell-core", out_sink);
             auto clean_env = get_modified_clean_environment(base_env.env_map, powershell_exe_path.parent_path());
-            if (build_env_cmd.empty())
-                return clean_env;
-            else
-                return cmd_execute_and_capture_environment(build_env_cmd, clean_env);
+            auto action_env =
+                build_env_cmd.empty() ? clean_env : cmd_execute_and_capture_environment(build_env_cmd, clean_env);
+            for (const auto& env_setup_script : pre_build_info.environment_setup_scripts)
+            {
+                const auto env_setup_cmd = make_setup_env_cmd(paths, env_setup_script);
+                if (vcpkg::Strings::ends_with(env_setup_script, ".cmake"))
+                {
+                    ProcessLaunchSettings settings;
+                    settings.environment = action_env;
+                    cmd_execute(env_setup_cmd, settings);
```

Comment:
Ah ok I need to actually look at the return value here. 
Feel like `cmd_execute` should be `[[no_discard]]`